nifi-commons/nifi-utils/src/main/java/org/apache/nifi/stream/io/util/TextLineDemarcator.java

```diff
@@ -95,52 +71,61 @@ public OffsetInfo nextOffsetInfo() {
      *
      * @return offset info
      */
-    public OffsetInfo nextOffsetInfo(byte[] startsWith) {
+    public OffsetInfo nextOffsetInfo(byte[] startsWith) throws IOException {
         OffsetInfo offsetInfo = null;
-        int lineLength = 0;
-        byte[] token = null;
-        lineLoop:
-        while (this.bufferLength != -1) {
+        byte previousByteVal = 0;
+        byte[] data = null;
+        nextTokenLoop: 
+        while (data == null && this.bufferLength != -1) {
             if (this.index >= this.bufferLength) {
                 this.fill();
             }
+            int delimiterSize = 0;
             if (this.bufferLength != -1) {
-                int i;
                 byte byteVal;
+                int i;
                 for (i = this.index; i < this.bufferLength; i++) {
                     byteVal = this.buffer[i];
-                    lineLength++;
-                    int crlfLength = computeEol(byteVal, i + 1);
-                    if (crlfLength > 0) {
-                        i += crlfLength;
-                        if (crlfLength == 2) {
-                            lineLength++;
-                        }
-                        offsetInfo = new OffsetInfo(this.offset, lineLength, crlfLength);
+
+                    if (byteVal == 10) {
```

Comment:
Can we get rid of inline constants 10 & 13 here and instead make them static final member variables to make the code a little more readable?